src/main/java/junmin/netflix/costreportservice/service/EPCostReportService.java

```diff
@@ -0,0 +1,169 @@
+package junmin.netflix.costreportservice.service;
+
+import junmin.netflix.costreportservice.exception.ErrorEnum;
+import junmin.netflix.costreportservice.exception.InvalidProductionException;
+import junmin.netflix.costreportservice.exception.InvalidReportException;
+//import junmin.netflix.costreportservice.model.EPItemizedCostEntity;
+import junmin.netflix.costreportservice.model.EPTotalCostEntity;
+import junmin.netflix.costreportservice.pojo.EPCostRecord;
+import junmin.netflix.costreportservice.repository.EPItemizedCostRepository;
+import junmin.netflix.costreportservice.repository.EPTotalCostRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.*;
+
+@Service
+public class EPCostReportService {
+	private EPTotalCostRepository epTotalCostRepo;
+	private EPItemizedCostRepository epItemizedCostRepo;
+
+	public EPCostReportService(EPTotalCostRepository epTotalCostRepo, EPItemizedCostRepository epItemizedCostRepo){
+		this.epTotalCostRepo = epTotalCostRepo;
+		this.epItemizedCostRepo = epItemizedCostRepo;
+	}
+
+	private final Logger LOGGER = LoggerFactory.getLogger(EPCostReportService.class);
+	private final String AMORIZEDCODE = "001";
+
+
+	public List<EPCostRecord> processEPCostReport(String prodName, List<EPCostRecord> costs) throws InvalidReportException
```

Comment:
Name is confusing.  processSingleEpisodeCostReport would be preferable.  Also, avoid acronyms/abbreviations.

Reply:
I tried to have a balance between clarity and length of names (for both methods or variables). :-) 
My other method is called processProductCostReport, so I was hoping this is clear enough to refer to Episode(single). 
I also searched up what's a good abbreviation for episode and think EP is a commonly recognized one. 
But this can be changed depends on team preference.

Reply:
> And we do not really need three POST methods, processProductCostReportWithAmortization is sufficient to handle all three scenarios. In the real design, I would just have one POST method. I have three because the doc asked me to have new endpoint for each scenario.

Good observation!  The first two cases are degenerate cases of the third.

You could have had three POST methods utilize a single implementation. That does make it a bit harder to reflect different types of errors for each endpoint, though.  For example, if you get an amortized cost in the two endpoints that don't 'officially' support them you might want to throw an exception.  There's some ambiguity in the specification as to what the desired behavior is.

FWIW I think if this were an actual project and not just a coding exercise someone would at some point have sad "Hey why don't we just make this one endpoint?"

Reply:
Agree on everything you said here!